src/dialects/mssql/query-interface.js

```diff
@@ -50,6 +50,29 @@ export class MsSqlQueryInterface extends AbstractQueryInterface {
     return this.sequelize.queryRaw(removeSql, options);
   }
 
+  /**
+    * @override
+    */
+  async bulkInsert(tableName, records, options, attributes) {
+    options = { ...options, type: QueryTypes.INSERT };
+
+    // If more than 1,000 rows are inserted outside of a transaction, we can't guarantee safe rollbacks.
+    // See https://github.com/sequelize/sequelize/issues/15426
+    if ((records.length <= 1000) || options.transaction) {
+      const sql = this.queryGenerator.bulkInsertQuery(tableName, records, options, attributes);
+
+      // unlike bind, replacements are handled by QueryGenerator, not QueryRaw
+      delete options.replacements;
+
+      const results = await this.sequelize.queryRaw(sql, options);
+
+      return results[0];
+    }
+
+    throw new Error('MSSQL doesn\'t allow for inserting more than 1,000 rows at a time. Please run this in a transaction to ensure safe rollbacks');
```

Comment:
Use a template string to avoid needing to escape `'`.

I think this error message should mention that because MSSQL doesn't allow inserting more than 1k rows, *we need to execute the insert as multiple queries*